File: src/main/java/ee/carlrobert/codegpt/actions/toolwindow/ClearChatWindowAction.java
package ee.carlrobert.codegpt.actions.toolwindow;

import com.intellij.icons.AllIcons;
import com.intellij.openapi.actionSystem.ActionUpdateThread;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.project.DumbAwareAction;
import ee.carlrobert.codegpt.actions.ActionType;
import ee.carlrobert.codegpt.actions.editor.EditorActionsUtil;
import ee.carlrobert.codegpt.telemetry.TelemetryAction;
import org.jetbrains.annotations.NotNull;

public class ClearChatWindowAction extends DumbAwareAction {

  private final Runnable onActionPerformed;

  public ClearChatWindowAction(Runnable onActionPerformed) {
    super("Clear Window", "Clears a chat window", AllIcons.General.Reset);
    this.onActionPerformed = onActionPerformed;
    EditorActionsUtil.registerAction(this);
  }

  @Override
  public void actionPerformed(@NotNull AnActionEvent event) {
    try {
      onActionPerformed.run();
    } finally {
      TelemetryAction.IDE_ACTION.createActionMessage()
          .property("action", ActionType.CLEAR_CHAT_WINDOW.name())
          .send();
    }
  }

  @Override
  public @NotNull ActionUpdateThread getActionUpdateThread() {
    return ActionUpdateThread.BGT;
  }
}

File: src/main/kotlin/ee/carlrobert/codegpt/actions/toolwindow/ClearChatWindowAction.kt
package ee.carlrobert.codegpt.actions.toolwindow

import com.intellij.icons.AllIcons
import com.intellij.openapi.actionSystem.ActionUpdateThread
import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import ee.carlrobert.codegpt.actions.ActionType
import ee.carlrobert.codegpt.actions.editor.EditorActionsUtil.registerAction
import ee.carlrobert.codegpt.conversations.ConversationsState.getCurrentConversation
import ee.carlrobert.codegpt.telemetry.TelemetryAction

class ClearChatWindowAction(private val onActionPerformed: Runnable) :
  AnAction("Clear Window", "Clears a chat window", AllIcons.General.Reset) {
  init {
    registerAction(this)
  }

  override fun update(event: AnActionEvent) {
    super.update(event)
    event.presentation.isEnabled = getCurrentConversation()?.messages?.isNotEmpty() == true
  }

  override fun actionPerformed(event: AnActionEvent) {
    try {
      onActionPerformed.run()
    } finally {
      TelemetryAction.IDE_ACTION.createActionMessage()
        .property("action", ActionType.CLEAR_CHAT_WINDOW.name)
        .send()
    }
  }

  override fun getActionUpdateThread(): ActionUpdateThread {
    return ActionUpdateThread.BGT
  }
}
